### ISSUE TITLE
Repeatability of results

### ISSUE THREAD

**Issue by 863689877** [sniklaus/softmax-splatting#56]
Hi, I use your softsplat function.
I found that the same input gives slightly different results each time.
Have you encountered this problem, how to solve it?

**Comment by sniklaus** [comment]
If I run the following I am always getting something in the range of 1e-9 so I am afraid that I am not seeing the behavior that you are describing. In other words, if I call softsplat twice with the same input I essentially get the same output. Can you share code that reproduces your issue?

```
import torch

import softsplat # the custom softmax splatting layer

tenImage = torch.randn(2, 3, 10, 20).cuda()
tenFlow = torch.randn(2, 2, 10, 20).cuda()
tenMetric = torch.randn(2, 1, 10, 20).cuda()

tenOne = softsplat.softsplat(tenIn=tenImage, tenFlow=tenFlow, tenMetric=tenMetric, strMode='soft')
tenTwo = softsplat.softsplat(tenIn=tenImage, tenFlow=tenFlow, tenMetric=tenMetric, strMode='soft')

print((tenOne - tenTwo).abs().mean().item())
```

**Comment by 863689877** [comment]
The key point of the problem is that you get 1e-9, not the exact 0.
This error will result in accumulation and amplification at the back layers of a DNN.

**Comment by sniklaus** [comment]
You might be in for a rude awakening, this prints 1e-7 even though it is doing mathematically the same thing.

```
import torch

netIn = torch.randn(32, 64, 128, 128).cuda()
netConv = torch.nn.Conv2d(64, 64, 3).cuda()

torch.backends.cudnn.enabled = True
torch.backends.cudnn.benchmark = True

tenOne = netConv(netIn)

torch.backends.cudnn.enabled = False
torch.backends.cudnn.benchmark = False

tenTwo = netConv(netIn)

print((tenOne - tenTwo).abs().mean().item())
```

**Comment by sniklaus** [comment]
Closing due to inactivity, feel free to reopen if this is still an issue with our work. :+1:
